plugins/redis/README

```diff
@@ -0,0 +1,124 @@
+use the asynchronous hiredis API, and 'sync' the synchronous API. Due to the nature of ECL as a declarative
+and non-functional language, no async activity breaches directly into the ECL layer in a usable manner, i.e.
+callbacks can not be used by the ECL activities EVENT and/OR WAIT. This being said, due to the current
+constraints the ECL plugin infrastructure, some async logic does actually surface into the ECL layer but is
+not directly accessible by any ECL, other than via these plugin functions. Such logic only concerns the
+pub/sub nature of redis based around a locking mechanism on keys.
+
+A full list of redis commands can be found at http://redis.io/commands of which the following are directly
+wrapped in the sync module only:
+
+BOOLEAN Exists(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+FlushDB(CONST VARSTRING options, UNSIGNED database = 0)
+Del(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+Persist(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+Expire(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED4 expire, UNSIGNED database = 0)
+INTEGER DBSize(CONST VARSTRING options, UNSIGNED database = 0)
+
+The redis command 'INFO ALL' is called once per instance of server:port combination and is written to the log
+file (sync only). Only sync connections to a single redis server:port are cached internally, changes in db are
+handled internally by the redis command 'SELECT DB'.
+
+The core SET functions have the prototype pattern (e.g.)
+SetString(CONST VARSTRING options, CONST VARSTRING key, CONST STRING value, UNSIGNED database = 0,  UNSIGNED4 expire = 0 /*days*/)
+The core GET functions have the prototype pattern (e.g.)
+STRING GetString(CONST VARSTRING options, CONST VARSTRING key, UNSIGNED database = 0)
+'options' is the redis server ip:port in the format '--SERVER=<ip>:<port>'. This defaults to localhost:6379 if
+omitted.
+
+This pattern differs in the 'locking' module taking the following forms:
+
+STRING SetString(UNSIGNED8 keyPtr, CONST STRING value, UNSIGNED4 expire = 0)
+STRING GetString(UNSIGNED8 keyPtr)
+
+Note that SET also returns a value. This is the vaue just set and not a wrapper for the redis command
+'GETSET'. See the section on locks.
+
+Within the sync and async modules, SETs overwrite their key values. The respective GETs (other than for
+STRINGs) will fail upon reuqest for a non-existent key. If you're not sure use EXISTS first (certain
+precaution has to be taken when using multiple plugin calls within ECL, regarding their actual execution
+timing. Remember ECL is declarative and may decide to hoist your calls.) GetString will return an empty
+STRING upon a cache miss.
+
+/**********************/
+/*     Locking        */
+/**********************/
+The locking module was designed with a particular use in mind. This concerns the scenario where multiple
```

Comment:
This section could do with revisiting.  In some places the text could be simplified.  Also good to making sure it is clear from the point of view of someone who wants to use the system, and doesn't particularly care how it is implemented.